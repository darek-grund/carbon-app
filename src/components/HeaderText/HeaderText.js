import React from 'react';
import PropTypes from 'prop-types';
import StyledHeaderText from './HeaderText.style';

const HeaderText = (props) => {
  return <StyledHeaderText>{props.text}</StyledHeaderText>;
};

HeaderText.propTypes = {
  text: PropTypes.string
};

export default HeaderText;
