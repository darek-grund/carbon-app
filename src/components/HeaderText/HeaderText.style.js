import styled from 'styled-components';

const StyledHeaderText = styled.div`
  padding: 30px;
  text-align: center;
  font-size: 20px;
  font-weight: 700;
  background-color: ${props => props.theme.bg};
  color: ${props => props.theme.primary};
`;

export default StyledHeaderText;
