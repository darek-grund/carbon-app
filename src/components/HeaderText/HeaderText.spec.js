import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import HeaderText from './HeaderText';

// Demonstration of a simple test for the HelloWorld component:
describe('<HeaderText />', () => {
  it('renders as expected', () => {
    const wrapper = renderer.create(<HeaderText />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders text', () => {
    const text = 'just random text';
    const wrapper = shallow(<HeaderText text={ text } />);
    expect(wrapper.contains(text)).toBe(true);
  });
});
