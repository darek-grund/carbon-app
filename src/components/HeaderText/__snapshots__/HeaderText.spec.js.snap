// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderText /> renders as expected 1`] = `
.c0 {
  padding: 30px;
  text-align: center;
  font-size: 20px;
  font-weight: 700;
}

<div
  className="c0"
/>
`;
